src/pfs/pfs.proto

```diff
@@ -123,7 +151,40 @@ message Commit {
   Branch branch = 1;
 }
 
-// CommitInfo is the main data structure representing a commit in etcd
+// CommitPicker defines mutually exclusive pickers that resolve to a single commit.
+// Commits can be picked relatively from some other commit like a parent or start of branch.
+// Alternatively, they can be picked via their global Id, which is composed of a repo picker and an id.
+// Picker messages should only be used as request parameters.
+message CommitPicker {
+  message CommitByGlobalId {
+    RepoPicker repo = 1;
+    string id = 2;
+  }
+
+  // TODO: add a picker option to pick commit by uuid.
+  // the uuid could be generated using AES so that it doesn't have to be stored.
+
+  // This models .N syntax.
+  message BranchRoot {
+    uint32 offset = 1;
```

Comment:
This is necessary because it's the opposite direction from `AncestorOf`?

Reply:
Yes and its a feature we have here:
https://docs.pachyderm.io/latest/learn/glossary/ancestry-syntax/